src/ServiceControl/CompositeViews/Endpoints/KnownEndpointUpdateHandler.cs

```diff
@@ -8,24 +8,34 @@ namespace ServiceControl.CompositeViews.Endpoints
 
     public class KnownEndpointUpdateHandler : IHandleMessages<EnableEndpointMonitoring>, IHandleMessages<DisableEndpointMonitoring>
     {
-        public IDocumentSession Session { get; set; }
+        private readonly IBus bus;
+        private readonly IDocumentStore store;
 
-        public IBus Bus { get; set; }
+        public KnownEndpointUpdateHandler(IBus bus, IDocumentStore store)
+        {
+            this.bus = bus;
+            this.store = store;
+        }
 
         public void Handle(EnableEndpointMonitoring message)
         {
-            var knownEndpoint = Session.Load<KnownEndpoint>(message.EndpointId);
+            KnownEndpoint knownEndpoint;
 
-            if (knownEndpoint == null)
+            using (var session = store.OpenSession())
             {
-                throw new Exception("No endpoint with found with id: " + message.EndpointId);
-            }
+                knownEndpoint = session.Load<KnownEndpoint>(message.EndpointId);
+
+                if (knownEndpoint == null)
+                {
+                    throw new Exception($"No endpoint with found with id: {message.EndpointId}");
+                }
 
-            knownEndpoint.Monitored = true;
+                knownEndpoint.Monitored = true;
```

Comment:
I wonder if this shouldn't just be a patch request. It seems silly to do a load-update-save cycle to update a single boolean flag